plugins/services/src/js/utils/DeclinedOffersUtil.js

```diff
@@ -0,0 +1,79 @@
+import DeclinedOffersReasons from '../constants/DeclinedOffersReasons';
+
+const DecinedOffersUtil = {
+  getSummaryFromQueue(queue) {
+    const {app, lastUnusedOffers, processedOffersSummary} = queue;
+
+    if (processedOffersSummary == null
+      || processedOffersSummary.unusedOffersCount === 0
+      || lastUnusedOffers == null
+      || lastUnusedOffers.length === 0) {
+      return null;
+    }
+
+    const {processedOffersCount = 0, rejectReason = {}} = processedOffersSummary;
+    const matchedRoleOffers = processedOffersCount -
```

Comment:
The API for this has changed, so we don't have to do this funnel-down math anymore (it all happens in the backend). We still have to do some math to figure out the number of matched offers, but it's more simple.

We take the number of processed offers and subtract the number of declined offers to get the number of matched offers.